clients/src/main/java/org/apache/kafka/clients/consumer/MockConsumer.java

```diff
@@ -477,11 +495,8 @@ private void resetOffsetPosition(TopicPartition tp) {
         seek(tp, offset);
     }
 
-    private Long getEndOffset(List<Long> offsets) {
-        if (offsets == null || offsets.isEmpty()) {
-            return null;
-        }
-        return offsets.size() > 1 ? offsets.remove(0) : offsets.get(0);
+    private MockConsumerLog<K, V> getLog(TopicPartition tp) {
```

Comment:
If we exposed this, maybe we could get rid of all the forwarding methods (such as `updateBeginningOffsets` and `truncatePartition`?